src/tallies/tally_scoring.cpp

```diff
@@ -0,0 +1,2347 @@
+        score = flux;
+      }
+      break;
+
+
+    case SCORE_TOTAL:
+      if (tally.estimator_ == ESTIMATOR_ANALOG) {
+        // All events will score to the total reaction rate. We can just use
+        // use the weight of the particle entering the collision as the score
+        if (settings::survival_biasing) {
+          // We need to account for the fact that some weight was already
+          // absorbed
+          score = (p->wgt_last_ + p->wgt_absorb_) * flux;
+        } else {
+          score = p->wgt_last_ * flux;
+        }
+
+      } else {
+        if (i_nuclide >= 0) {
+          score = p->neutron_xs_[i_nuclide].total * atom_density * flux;
+        } else {
+          score = p->macro_xs_.total * flux;
+        }
+      }
+      break;
+
+
+    case SCORE_INVERSE_VELOCITY:
+      if (tally.estimator_ == ESTIMATOR_ANALOG) {
+        // All events score to an inverse velocity bin. We actually use a
+        // collision estimator in place of an analog one since there is no way
+        // to count 'events' exactly for the inverse velocity
+        if (settings::survival_biasing) {
+          // We need to account for the fact that some weight was already
+          // absorbed
+          score = p->wgt_last_ + p->wgt_absorb_;
+        } else {
+          score = p->wgt_last_;
+        }
+        score *= flux / p->macro_xs_.total;
+      } else {
+        score = flux;
+      }
+      // Score inverse velocity in units of s/cm.
+      score /= std::sqrt(2. * E / MASS_NEUTRON_EV) * C_LIGHT * 100.;
+      break;
+
+
+    case SCORE_SCATTER:
+      if (tally.estimator_ == ESTIMATOR_ANALOG) {
+        // Skip any event where the particle didn't scatter
+        if (p->event_ != EVENT_SCATTER) continue;
+        // Since only scattering events make it here, again we can use the
+        // weight entering the collision as the estimator for the reaction rate
+        score = p->wgt_last_ * flux;
+      } else {
+        if (i_nuclide >= 0) {
+          score = (p->neutron_xs_[i_nuclide].total
+            - p->neutron_xs_[i_nuclide].absorption) * atom_density * flux;
+        } else {
+          score = (p->macro_xs_.total
+            - p->macro_xs_.absorption) * flux;
+        }
+      }
+      break;
+
+
+    case SCORE_NU_SCATTER:
+      // Only analog estimators are available.
+      // Skip any event where the particle didn't scatter
+      if (p->event_ != EVENT_SCATTER) continue;
+      // For scattering production, we need to use the pre-collision weight
+      // times the yield as the estimate for the number of neutrons exiting a
+      // reaction with neutrons in the exit channel
+      if (p->event_mt_ == ELASTIC || p->event_mt_ == N_LEVEL ||
+        (p->event_mt_ >= N_N1 && p->event_mt_ <= N_NC)) {
+        // Don't waste time on very common reactions we know have
+        // multiplicities of one.
+        score = p->wgt_last_ * flux;
+      } else {
+        // Get yield and apply to score
+        auto m =
```

Comment:
Optional: Recommend another variable name other than `m`.  It's hyper local (used on the next line), but it's not 100% clear what the variable has from context, which can open the code up to subtle bugs when refactoring.